### ISSUE TITLE
bring back OG flashlight

### ISSUE THREAD

**Issue by Die4Ever** [KozGit/DOOM-3-BFG-VR#285]
this might be difficult, but I think the original Doom 3 where you had to switch between the gun and the flashlight made the game way more intense, this is what the levels were designed for

having the flashlight always available removes the gameplay choice of flashlight vs gun, and that kills the tension

also hitting things with the flashlight was fun

and BFG Edition's new armor-mounted flashlight doesn't even cast shadows

**Comment by CarlKenner** [comment]
You're right, we should add an option for that. And also add the opposite option for people who want a strictly BFG-Edition-style armour-mounted flashlight that you can't move.

The armour-mounted flashlight in our mod should already cast shadows, but I need to check.

Currently, you can choose in the menu whether to have your flashlight mounted on your helmet, your armour, or your non-weapon hand, and you can also manually pick-up the flashlight and transfer it between those places by using the grip button in those locations. But even when you are holding it in your hand, you can't currently use it as a weapon.

We're in the process of adding dual-wielding, so you can hold a weapon in each hand. We are adding a setting to specify how much dual-wielding the player should be able to do, so I guess we should add an even lower setting where you can't even dual-wield the flashlight and a weapon.

Then we also need to make it function as a weapon, and add a setting so it optionally doesn't run out of batteries, so that way it will be like the original Doom 3.

BTW, the original Doom3: Resurrection of Evil for the original XBox had the flashlight permanently mounted on the pistol, so we should add an option for that too. https://youtu.be/HdVAjlRuWM4?t=22

**Comment by Die4Ever** [comment]
these sound like great options, I can't wait to play Doom 3 in VR with the flashlight as originally intended!

**Comment by CarlKenner** [comment]
You're right, the armour-mounted flashlight doesn't cast a shadow.

**Comment by CarlKenner** [comment]
I've fixed it so you can now optionally disable dual-wielding flashlights, and you can kind-of cycle through the flashlight like a weapon. It still doesn't work as a weapon though, and the cycling is still a bit weird because it swaps your dominant-hand.
And I got the RoE (XBox) pistol-mounted flashlight mostly working (I just need to mount it in the correct place on the weapon now).
So I'm making progress.
